src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DevirtualizationManager.cs

```diff
@@ -22,6 +23,10 @@ public override bool IsEffectivelySealed(TypeDesc type)
 
         public override bool IsEffectivelySealed(MethodDesc method)
         {
+            if (method.IsUnboxingThunk())
+            {
+                return IsEffectivelySealed(method.GetUnboxedMethod());
+            }
```

Comment:
I'd do this in JitInterface - unboxing thunks are supposed to be local to JitInterface - they should never escape it (having to add `using Internal.JitInterface` means they escaped).